allure-report-data/src/test/groovy/ru/yandex/qatools/allure/data/converters/DefaultTestCaseConverterTest.groovy

```diff
@@ -296,6 +299,90 @@ class DefaultTestCaseConverterTest {
         assert modify.attachments[0].uid
     }
 
+    @Test
+    void shouldConvertTestCaseWithoutNameAndTitle() {
+        def origin = new TestCaseResult()
+        def modify = converter.convert(origin)
+
+        assert modify.name
+        assert modify.name == UNKNOWN_TEST_CASE
+
+        assert modify.title
+        assert modify.title == TextUtils.humanize(UNKNOWN_TEST_CASE)
+    }
+
+    @Test
+    void shouldConvertTestCaseWithoutName() {
+        def origin = new TestCaseResult(title: "some title")
+        def modify = converter.convert(origin)
+
+        assert modify.name
+        assert modify.name == UNKNOWN_TEST_CASE
+
+        assert modify.title
+        assert modify.title == "some title"
+    }
+
+    @Test
+    void shouldConvertTestCaseWithoutTitle() {
+        def origin = new TestCaseResult(name: "someName")
+        def modify = converter.convert(origin)
+
+        assert modify.name
+        assert modify.name == "someName"
+
+        assert modify.title
+        assert modify.title == "Some name"
+    }
+
+    @Test
+    void shouldConvertStepWithoutNameAndTitle() {
+        def origin = new TestCaseResult(
+                name: "name",
+                steps: [new Step()]
+        )
+        def modify = converter.convert(origin)
+
+        assert modify.steps
+        assert modify.steps.size() == 1
+        assert modify.steps[0].name
+        assert modify.steps[0].name == UNKNOWN_STEP_NAME
+        assert modify.steps[0].title
+        assert modify.steps[0].title == TextUtils.humanize(UNKNOWN_STEP_NAME)
+    }
+
+    @Test
+    void shouldConvertStepWithoutName() {
+        def origin = new TestCaseResult(
+                name: "name",
+                steps: [new Step(title: "some title")]
+        )
+        def modify = converter.convert(origin)
+
+        assert modify.steps
```

Comment:
@baev I would agree with @lanwen. JUnit best practice is to use matchers. Either don't use JUnit in these tests which is not a good idea or migrate to matchers. However you could use Groovy style calls without parentheses. Something like:
```groovy
assertThat loaded hasSize 3
```